internal/objects/service/service.go

```diff
@@ -240,13 +244,16 @@ func DesiredEnvoyService(contour *operatorv1alpha1.Contour) *corev1.Service {
 		},
 	}
 
-	// Add the NLB annotation if specified by AWS provider parameters.
-	if nlbNeeded(&contour.Spec) {
-		svc.Annotations[awsLBTypeAnnotation] = "nlb"
-	}
-	// Add the TCP backend protocol annotation for AWS classic load balancers.
-	if backendTCPNeeded(&contour.Spec) {
-		svc.Annotations[awsLbBackendProtoAnnotation] = "tcp"
+	// Add AWS LB annotations based on the network publishing strategy and provider type.
+	if contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.LoadBalancerServicePublishingType {
+		// Add the TCP backend protocol annotation for AWS classic load balancers.
+		if isELB(&contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters) {
+			svc.Annotations[awsLbBackendProtoAnnotation] = "tcp"
+			svc.Annotations[awsLBProxyProtocolAnnotation] = "*"
+		} else {
+			// Annotate the service for an NLB.
+			svc.Annotations[awsLBTypeAnnotation] = "nlb"
+		}
```

Comment:
should we be checking for `params.Type == operatorv1alpha1.AWSLoadBalancerProvider` here to only add the annotation if AWS is requested? It looks like the previous code did that. Worth adding some tests for too.

Reply:
@skriss thanks for catching this. Commit `bbfec50` fixes this issue, PTAL.